component/authenticator/src/test/java/org/wso2/carbon/identity/application/authenticator/totp/util/TOTPUtilTest.java

```diff
@@ -601,4 +605,48 @@ public IObjectFactory getObjectFactory() {
 
         return new PowerMockObjectFactory();
     }
+
+    @Test(description = "Test case for getProcessedClaimValue()", dataProvider = "processedClaimValueTestDataProvider")
+    public void testGetProcessedClaimValue(String claimURI, List<LocalClaim> localClaims, String claimValue,
+                                           String expectedClaimValue) throws Exception {
+
+        mockStatic(TOTPDataHolder.class);
+        when(totpDataHolder.getInstance()).thenReturn(totpDataHolder);
+        when(totpDataHolder.getClaimMetadataManagementService()).thenReturn(claimMetadataManagementService);
+        when(claimMetadataManagementService.getLocalClaims("testDomain")).thenReturn(localClaims);
+        assertEquals(TOTPUtil.getProcessedClaimValue(claimURI, claimValue,"testDomain"), expectedClaimValue);
+    }
+
+    @DataProvider(name = "processedClaimValueTestDataProvider")
+    public static Object[][] getProcessedClaimValueTestData() {
+
+        HashMap<String, String> claimProperties_1 = new HashMap<>();
+        claimProperties_1.put("EnableEncryption", "true");
+        List<LocalClaim> localClaims1 = new ArrayList<>();
+        LocalClaim localClaim1 = new LocalClaim("http://wso2.org/claims/identity/secretkey", null,
+                claimProperties_1);
+        localClaims1.add(localClaim1);
+
+        HashMap<String, String> claimProperties_2 = new HashMap<>();
+        claimProperties_2.put("EnableEncryption", "false");
+        List<LocalClaim> localClaims2 = new ArrayList<>();
+        LocalClaim localClaim2 = new LocalClaim("http://wso2.org/claims/identity/secretkey", null,
+                claimProperties_2);
+        localClaims2.add(localClaim2);
+
+        HashMap<String, String> claimProperties_3 = new HashMap<>();
+        List<LocalClaim> localClaims3 = new ArrayList<>();
+        LocalClaim localClaim3 = new LocalClaim("http://wso2.org/claims/identity/secretkey", null,
+                claimProperties_3);
+        localClaims3.add(localClaim3);
+
+        return new Object[][]{
+                {"http://wso2.org/claims/identity/secretkey", localClaims1, "AER2BRI0LK4XCSC1", "AER2BRI0LK4XCSC1"},
+                {"http://wso2.org/claims/identity/secretkey", localClaims1, "", ""},
+                {"http://wso2.org/claims/identity/secretkey", localClaims2, "AER2BRI0LK4XCSC1", "AER2BRI0LK4XCSC1"},
+                {"http://wso2.org/claims/identity/secretkey", localClaims2, "", ""},
+//                {"http://wso2.org/claims/identity/secretkey", localClaims3, "AER2BRI0LK4XCSC1", null},
+//                {"http://wso2.org/claims/identity/secretkey", localClaims3, "", ""},
```

Comment:
why are these commented?

Reply:
will be addressed in https://github.com/wso2-extensions/identity-outbound-auth-totp/pull/162